client/my-sites/site-settings/search.jsx

```diff
@@ -266,7 +267,9 @@ const checkForSearchProduct = ( purchase ) =>
 export default connect( ( state, { isRequestingSettings } ) => {
 	const site = getSelectedSite( state );
 	const siteId = getSelectedSiteId( state );
-	const hasSearchProduct = getSitePurchases( state, siteId ).find( checkForSearchProduct );
```

Comment:
We are only checking if the site has Jetpack Search and not if the site's plan includes Jetpack Search.